sympy/core/mod.py

```diff
@@ -64,7 +62,7 @@ def doit(p, q):
             except TypeError:
                 pass
             else:
-                if type(d) is int:
+                if type(d) is int or type(d) is long:
```

Comment:
perhaps add not here `# when 2.7 is dropped, the long clause will not be needed`

Reply:
This could be `if type(d) in integer_types` (from core.compatibility). Or `isinstance(d, integer_types)`.